### PR TITLE
cargo-insta: use cargo integration tests

### PR DESCRIPTION
**Pull request by tamird**
@mitsuhiko this should be able to land independently of #410.

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -8,11 +8,7 @@ use std::{
 
 /// Are we running in in a CI environment?
 pub fn is_ci() -> bool {
-    match env::var("CI").ok().as_deref() {
-        Some("false") | Some("0") | Some("") => false,
-        None => env::var("TF_BUILD").is_ok(),
-        Some(_) => true,
-    }
+    env::var_os("CI").is_some() || env::var_os("TF_BUILD").is_some()
```
Comment (mitsuhiko) [inline src/utils.rs:11]: What is the motivation of this change? This now means that it cannot be disabled any more by setting to an empty string.

Comment (tamird) [reply src/utils.rs]: I think I stumbled upon this when I had to change the `cd` dance in the Makefile, but that's no longer correct until #410 lands. Reverted.

**Comment by mitsuhiko** [comment]
What confuses me is how the rustfmt fails to run now with these changes on 1.61.

**Comment by tamird** [comment]
Turns out using cargo integration tests here is more complicated than I had anticipated. I'm going to do something simpler than originally intended in #410 and come back to this later.

**Comment by tamird** [comment]
@mitsuhiko this is ready for another look. The first two commits are minor CI improvements. I'm not sure the last commit is a win, but I wanted to revisit this after finally understanding how these integration tests work.
